Tasks/tests/System/V2beta2/TasksServiceSmokeTest.php

```diff
@@ -0,0 +1,91 @@
+<?php
+/*
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+namespace Google\Cloud\Tasks\Tests\System\V2beta2;
+
+use Google\Cloud\Core\ExponentialBackoff;
+use Google\Cloud\Core\Testing\System\SystemTestCase;
+use Google\Cloud\Tasks\V2beta2\CloudTasksClient;
+use Google\Cloud\Tasks\V2beta2\LeaseDuration;
+use Google\Cloud\Tasks\V2beta2\PullMessage;
+use Google\Cloud\Tasks\V2beta2\PullTarget;
+use Google\Cloud\Tasks\V2beta2\Queue;
+use Google\Cloud\Tasks\V2beta2\Task;
+use Google\Cloud\Tasks\V2beta2\Task_View;
+use Google\Protobuf\Duration;
+
+/**
+ * @group tasks
+ * @group gapic
+ */
+class TasksServiceSmokeTest extends SystemTestCase
+{
+
+    public function createQueue($client, $locationName, $queue)
+    {
+        $backoff = new ExponentialBackoff(8);
+        $backoff->execute(function () use ($client, $locationName, $queue) {
+            $client->createQueue($locationName, $queue);
+        });
+        self::$deletionQueue->add(function () use ($backoff, $client, $queue) {
+            $backoff->execute(function () use ($client, $queue) {
+                $client->deleteQueue($queue->getName());
+            });
+        });
+        sleep(120);
```

Comment:
Instead of sleeping for a predetermined amount of time, is it possible to check [the state of the queue](https://github.com/GoogleCloudPlatform/google-cloud-php/pull/1123/files#diff-93f746032fc84d5fe3a66ee286751ed3R81) to continue with processing?